src/components/PlayerIcon/PlayerIcon.tsx

```diff
@@ -27,48 +27,38 @@ interface PlayerIconProps {
     className?: string;
 }
 
-export function icon_size_url(url, size) {
+export function icon_size_url(url: string, size: number): string {
     return url.replace(/-[0-9]+.png$/, `-${size}.png`).replace(/s=[0-9]+/, `s=${size}`);
 }
 
-export function getPlayerIconURL(id, size): Promise<string> {{{
-    return new Promise((resolve, reject) => {
-        player_cache.fetch(id, ["icon"]).then((user) => {
-            resolve(icon_size_url(user.icon, size));
-        })
-        .catch(reject);
-    });
-}}}
+export function getPlayerIconURL(id: number, size: number): Promise<string> {
+    return player_cache.fetch(id, ["icon"]).then(user => icon_size_url(user.icon, size));
```

Comment:
This does not do the same thing as the function you replaced: https://github.com/online-go/online-go.com/pull/400/files#diff-4a58653571e8b38e193938055200ebfcL34 and the types don't match up.. not sure why this compiles tbh